app/src/editor/shared/components/StreamSelector.jsx

```diff
@@ -63,32 +80,45 @@ export default class StreamSelector extends React.Component<Props, State> {
             uiChannel: false,
             public: true,
         }
+
         const streams = await getStreams(params)
 
-        if (this.unmounted) { return }
+        if (this.unmounted || this.currentSearch !== search) { return }
 
         this.setState({
             matchingStreams: streams,
         })
+    }, 500)
+
+    search = async (search: string) => {
+        this.currentSearch = search
+        search = search.trim()
```

Comment:
Fiddled around with it briefly. Changing streams seems to work fine. This line here explodes:

<img width="741" alt="Screenshot 2019-05-15 15 56 59" src="https://user-images.githubusercontent.com/320066/57781301-2f746600-772a-11e9-8150-206def991f0e.png">

To reproduce (1) search for `SendToStream` and add it to the canvas, then (2) focus the `Search` field – it explodes here.

Reply:
@mondoreale I cannot reproduce the problem, but I have seen something weird in a StreamSelector before that could be related… can you check does that StreamSelector have an `initialValue` property that's `{}`? I don't know how that `{}` got there but it would help to know it wasn't just something weird that happened to me during dev

Reply:
Never mind, it says in the stack trace it's `undefined`, not `{}`.

Reply:
I couldn't repro, but this should fix it fe9615f

Reply:
Here's what it takes for me to do it. It's gotta be a freshly added `SendToStream` module.

![May-16-2019 15-42-35](https://user-images.githubusercontent.com/320066/57858408-50ec5500-77f1-11e9-843c-2da97ec31deb.gif)

Now, I dug up something that IS connected. If the initial value is an empty string (which it is!) then `getStream(value)` turns into `getStream('')` and this is pretty much [`getStreams` from canvas services](https://github.com/streamr-dev/streamr-platform/blob/development/app/src/editor/canvas/services.js#L181-L183) w/o params.

We're expecting a stream, and receiving an array of streams (`[].name` is `undefined`).